### DISCUSSION TITLE
og-image-api/

### DISCUSSION THREAD

**Discussion by giscus** [eallion/eallion.com#103] [Giscus]
# og-image-api/

前言 Deprecated：我现在已经手动生成 OG Image 了。 DEMO: https://og.eallion.com/api/og?title=蜗牛 Vercel 官方有提

https://www.eallion.com/og-image-api/

**Comment by xkdvSrPD** [comment]
非常好的方案🤔，我在使用Nuxt时遇到了类似的需求，我的解决方案时部署一个无界面浏览器，对网站进行截图，从而生成Og Image。

具体方案：[https://vio.vin/article/use-puppeteer](https://vio.vin/article/use-puppeteer)

**Reply by eallion** [reply]
无头浏览器确实可以做很多事情。是个好方案。

**Comment by xinkeji** [comment]
emm,为了省事只准备改标题作者和背景图？，我还原了蜗牛大佬的背景图，是一个你的头像，并没有找到背景图，然后为了省事其他的都默认了，如字体什么的，原本只改头像可以正常使用，每次一改作者这类文字生成就会变成空白，不知道为什么

**Reply by eallion** [reply]
你是说这个 OG Image API 吗？那有可能是字体缺失，我压缩了只占我博客标题用到的字体。

**Comment by xinkeji** [comment]
😂😂了解了，我打算用来ai自动生成的时候图片也能自动生成，我主要还是嘀咕，偶尔写写长文，这个实在太适合我了，目前用大佬的源码，想问下那个自动摘要怎么调用json啊，我看你说不想破坏md文件，但是你就是改的md文件啊，我得看到了个插入摘要的脚本但是没有github action
文件啊，咋搞呀，大佬

---Original---
From: "Charles ***@***.***&gt;
Date: Fri, Mar 1, 2024 01:50 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [eallion/eallion.com] og-image-api/ (Discussion #103)




 
你是说这个 OG Image API 吗？那有可能是字体缺失，我压缩了只占我博客标题用到的字体。

**Reply by eallion** [reply]
原来用 json 文件存所有数据，用在另一个项目中，发现也有缺点，现在改成存在博客文章的 frontmatter 中了。没有用 CI/CD 生成，都是手动生成的，因为都是几个月才写一篇博客。另外要注意缓存，不然 token 消耗很快。

**Reply by xinkeji** [reply]
😂😂有啥缺点，如果没有大问题，可以教我下嘛
